pkg/cmd/history_writer.go

```diff
@@ -20,17 +20,28 @@ import (
 )
 
 // Write writes history to given path
-func (w *GardenctlHistoryWriter) Write(historyPath string, history map[string]string) error {
+func (w *GardenctlHistoryWriter) Write(historyPath string, m map[string]string) error {
 	f, err := os.OpenFile(historyPath, os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0666)
 	if err != nil {
 		return err
 	}
 
-	j, err := json.Marshal(history)
+	j, err := json.Marshal(m)
 	if err != nil {
 		return err
 	}
 
 	_, err = f.WriteString(string(j) + "\n")
 	return err
```

Comment:
why not call the newly introduced w.WriteString(..) here?
why is historyPath not a field on the GardenctlHistoryWriter struct

Reply:
* combine both WriteString and Write to one.
>why is historyPath not a field on the GardenctlHistoryWriter struct

actually, I don't have any particular reason  I just follow the same standard like others, but I was also think that before. the reasone I choose is just one line 

the `historyWriter    = &GardenctlHistoryWriter{}` init() in `root.go` 
then if I add historyPath filed, I need `historyWriter.HistoryPath = historyPath` and  `historyWriter.Write(targetInfo)`   in `target.go` and `target_history.go`   

or one line which I currently used in both files without `historyWriter.HistoryPath = historyPath`
`err := historyWriter.Write(pathHistory, targetInfo)`

https://github.com/gardener/gardenctl/blob/ca39a1ace8976fcaa0c013a878c225b9ee20f54c/pkg/cmd/types.go#L42-L52